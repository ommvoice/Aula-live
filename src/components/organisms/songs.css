.songs {
	max-width: 80%;
	margin: auto;
}
  
.song-card {
	box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
	transition: 0.3s;
	text-align: center;
	margin-bottom: 16px;
}

.container {
	padding: 2px 16px;
}

.loading {
	color: green;
}

.margin-btm{
	margin-bottom: 40px;
}

.text {
	display: block;
	width: 200px;
	overflow: hidden;
	white-space: nowrap;
	text-overflow: ellipsis;
}
