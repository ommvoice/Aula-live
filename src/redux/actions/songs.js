export const GET_SONGS = 'GET_SONGS';
export const RECEIVE_SONGS = 'RECEIVE_SONGS';
export const SEARCH_TEXT = 'SEARCH_TEXT';
export const ON_ERROR = 'ON_ERROR';

export function getSongs(category) {
  return {
    type: GET_SONGS,
    category,
  };
}

export function getSearch(search) {
  return {
    type: SEARCH_TEXT,
    search,
  };
}

export function receiveSongs(payload) {
  return {
    type: RECEIVE_SONGS,
    payload,
  };
}

export function onError() {
  return {
    type: ON_ERROR,
  };
}
