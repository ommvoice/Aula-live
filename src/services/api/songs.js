import { API, PROXY } from '../../constants';
import { get } from './axios';

export function songs(category, limit = 100) {
  const url = API + `/api/v1/us/apple-music/${category}/all/${limit}/non-explicit.json`;

  return get(PROXY + url);
}
