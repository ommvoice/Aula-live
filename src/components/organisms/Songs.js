import React from 'react';
import Heading from '../atoms/Heading';
import Select from '../atoms/Select';
import SongCard from '../molecules/SongCard';
import SearchBar from '../molecules/SearchBar';
import Loading from '../atoms/Loading';
import { SONG_CATEGORIES } from '../../constants';
import './songs.css';

const Songs = ({ seleceDefaultValue, searchText, onSearch, onSelect, songs }) => {
	return (
		<div className="songs">
			<div className="row">
				<Heading text='Select Songs Category' />
				<Select defaultValue={seleceDefaultValue} onChange={onSelect} options={SONG_CATEGORIES} />
			</div>
			<div className="row">
				<SearchBar value={searchText} label='Search' placeholder='Search By Artist Name Or Song Name' onChange={onSearch} />
			</div>
			<div className="row">
				<Heading text={`${songs.length} Results Found`} />
				{ songs.length ?
					songs.map((song, index) => (
						<SongCard key={index} imgsrc={song.artworkUrl100} text={song.artistName || song.name} />
					))
					: <Loading />
				}
			</div>
		</div>
	);
}

export default Songs;
