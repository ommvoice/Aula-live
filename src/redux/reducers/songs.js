import { RECEIVE_SONGS, SEARCH_TEXT, ON_ERROR } from '../actions/songs';

const initialState = {
  searchText: '',
  songs: [],
  error: null,
};

function receiveSongs(state = initialState, action) {
  switch (action.type) {
    case RECEIVE_SONGS:
      return {
        ...state,
        songs: action.payload && action.payload.data.feed.results
      };
    case SEARCH_TEXT:
      return {
        ...state,
        searchText: action.search
      };
    case ON_ERROR:
      return {
        ...state,
        error: true 
      };
    default:
      return state;
  }
}

export default receiveSongs;
