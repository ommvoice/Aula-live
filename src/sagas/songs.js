import { put, call, takeLatest } from 'redux-saga/effects';
import * as actions from '../redux/actions/songs';
import { songs } from '../services/api/songs';

export function* callSongsApi(param) {
  try {
    const result = yield call(songs, param && param.category || 'top-albums');
    yield put(actions.receiveSongs(result));
  } catch(error) {
    yield put(actions.onError(error));
  }
}

export function* receiveSongs() {
  yield takeLatest(actions.GET_SONGS, callSongsApi);
}
