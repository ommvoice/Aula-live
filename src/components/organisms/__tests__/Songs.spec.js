import React from 'react';
import { shallow } from 'enzyme';
import Songs from '../Songs';
import { SONGS } from '../../../mocks/fixtures/songs';

describe('Songs', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Songs songs={SONGS.data.feed.results} />);

    expect(wrapper).toMatchSnapshot();
  });
});
