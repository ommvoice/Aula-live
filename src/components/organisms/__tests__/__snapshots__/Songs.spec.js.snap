// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs renders correctly 1`] = `
<div
  className="songs"
>
  <div
    className="row"
  >
    <Heading
      text="Select Songs Category"
    />
    <Select
      options={
        Array [
          Object {
            "title": "Top Albums",
            "value": "top-albums",
          },
          Object {
            "title": "Top Songs",
            "value": "top-songs",
          },
          Object {
            "title": "Hot Tracks",
            "value": "hot-tracks",
          },
          Object {
            "title": "New Releases",
            "value": "new-releases",
          },
          Object {
            "title": "Coming Soon",
            "value": "coming-soon",
          },
        ]
      }
    />
  </div>
  <div
    className="row"
  >
    <SearchBar
      label="Search"
      placeholder="Search By Artist Name Or Song Name"
    />
  </div>
  <div
    className="row"
  >
    <Heading
      text="2 Results Found"
    />
    <SongCard
      key="0"
      text="Kodak Black"
    />
    <SongCard
      key="1"
      text="Anonymous"
    />
  </div>
</div>
`;
